addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthDiagnosticCollector.java

```diff
@@ -0,0 +1,278 @@
+        for (String value : header.getHeaderValues(headerName)) {
+            sb.append(headerName);
+            sb.append(": ");
+            // Special case
+            if (HttpHeader.AUTHORIZATION.equalsIgnoreCase(headerName)
+                    && value.toLowerCase().startsWith("bearer")) {
+                int offset = value.indexOf(" ");
+                if (offset == -1) {
+                    offset = value.indexOf(":");
+                }
+                if (offset == -1) {
+                    sb.append(getSanitizedToken(value));
+                } else {
+                    sb.append(value.substring(0, offset));
+                    sb.append(" ");
+                    sb.append(getSanitizedToken(value.substring(offset + 1)));
+                }
+            } else {
+                sb.append(getSanitizedToken(value));
+            }
+            sb.append("\n");
+        }
+    }
+
+    protected synchronized String getSanitizedHost(HttpMessage msg) throws URIException {
+        return getSanitizedHost(SessionStructure.getHostName(msg));
+    }
+
+    protected synchronized String getSanitizedHost(String host) {
+        if (!hostMap.containsKey(host)) {
+            hostMap.put(host, "https://example" + hostId++ + "/");
+        }
+        return hostMap.get(host);
+    }
+
+    protected synchronized String getSanitizedToken(String token) {
+        if (!tokenMap.containsKey(token)) {
+            tokenMap.put(token, "token" + tokenId++);
+        }
+        return tokenMap.get(token);
+    }
+
+    protected JSONObject sanitiseJson(JSONObject jsonObject) {
+        JSONObject sanObj = new JSONObject();
+        for (Object key : jsonObject.keySet()) {
+            Object val = jsonObject.get(key);
+            if (val instanceof String) {
+                sanObj.put(key, getSanitizedToken((String) val));
+            } else {
+                sanObj.put(key, val);
+            }
+        }
+        return sanObj;
+    }
+
+    protected Object sanitiseJson(Object obj) {
+        if (obj instanceof JSONObject) {
+            return sanitiseJson((JSONObject) obj);
+        } else if (obj instanceof JSONArray) {
+            JSONArray sanArr = new JSONArray();
+            Object[] oa = ((JSONArray) obj).toArray();
+            for (int i = 0; i < oa.length; i++) {
+                sanArr.add(sanitiseJson(oa[i]));
+            }
+            return sanArr;
+        } else if (obj instanceof String) {
+            return getSanitizedToken((String) obj);
+
+        } else {
+            return obj;
+        }
+    }
+
+    protected boolean isRelevant(HttpMessage msg) {
+        if (msg.getRequestHeader().isCss() || msg.getRequestHeader().isImage()) {
+            return false;
+        }
+        if (msg.getResponseHeader().isCss()
+                || msg.getResponseHeader().isImage()
+                || msg.getResponseHeader().isJavaScript()) {
+            return false;
+        }
```

Comment:
That confuses me. Looks like it might be relevant if request header is JavaScript but response header is something else.
Is there any chance to check the response header only for simplicity?

Reply:
We cant check the request header to see if its JavaScript - theres no such method. If there was I would have used it :)